chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
`clock` and `reset` are currently available in user-level Module code, right? Is this a change in that functionality? And if they are still available, what does the following code do:

``` 
class MyModule extends Module {
  
 val io = new Bundle { val otherClock = Clock(INPUT)); }

  WithClock(io.otherClock) {

  val subModule = Module(new SubModule())
  subModule.clock := clock

}
```

What clock does subModule actually get in the end? Or is something written above illegal now?